sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py

```diff
@@ -30,7 +31,7 @@
 
 
 logger = logging.getLogger(__name__)
-
+locale.setlocale(locale.LC_COLLATE, "")
```

Comment:
is this a global setting ?

Reply:
Yes, it is. I've been looking into ways to create some form of a locale object to sort based on without having to set globally.  So far, my findings have been: I can either run a child process and set the locale in it or set the locale right before my sort then undo it right after my sort (by resetting to the original locale).

Reply:
We should explore other ways to get sorting right. We shouldn't be setting globals like this as it may interfere with customers application.